src/rest_api.py

```diff
@@ -163,46 +163,38 @@ def user_repo_scan():
     """Experimental endpoint."""
     # TODO: please refactor this method is it would be possible to test it properly
     # json data and files cannot be a part of same request. Hence, we need to use form data here.
-    git_url = request.form.get('git-url')
-    ecosystem = request.form.get('ecosystem')
-
+    validate_string = "{} cannot be empty"
     resp_dict = {
         "status": "success",
         "summary": ""
     }
-
-    files = request.files.getlist("dependencyFile[]")
-
-    validate_string = "{} cannot be empty"
+    git_url = request.headers.get("git-url")
 
     if not git_url:
         validate_string = validate_string.format("git-url")
         resp_dict["status"] = 'failure'
         resp_dict["summary"] = validate_string
         return flask.jsonify(resp_dict), 400
 
-    if not files:
-        validate_string = validate_string.format("files")
+    req_json = request.json
+    set_direct_dependencies = set()
+    set_transitive_dependencies = set()
+    if req_json is None:
+        validate_string = validate_string.format("input json")
         resp_dict["status"] = 'failure'
         resp_dict["summary"] = validate_string
         return flask.jsonify(resp_dict), 400
 
-    if not ecosystem:
-        validate_string = validate_string.format("ecosystem")
+    result_ = req_json.get("result", None)
+    if result_ is None:
+        validate_string = validate_string.format("Result dictionary")
         resp_dict["status"] = 'failure'
         resp_dict["summary"] = validate_string
         return flask.jsonify(resp_dict), 400
 
-    try:
-        set_direct_dependencies, set_transitive_dependencies = \
-            get_parser_from_ecosystem(ecosystem).parse_output_files(files)
-    except BadRequest as b:
-        resp_dict['status'] = 'failure'
-        resp_dict['summary'] = b.get_body()
-        return flask.jsonify(resp_dict), 400
-
-    # we need to remove direct dependencies from the transitive ones.
-    set_transitive_dependencies = set_transitive_dependencies - set_direct_dependencies
+    for res_ in result_:
+        details_ = res_.get("details", None)
+        set_direct, set_transitive = DataExtractor.get_details_from_results(details_)
```

Comment:
you are storing it in set_direct and set_transitive whereas later you are trying to fetch from set_direct_dependencies. This will not work.